lib/MusicBrainz/Server/Data/Work.pm

```diff
@@ -64,31 +64,76 @@ sub _id_column
 
 sub find_by_artist
 {
-    my ($self, $artist_id, $limit, $offset) = @_;
+    my ($self, $artist_id, $limit, $offset, %args) = @_;
+
+    my (@where_query, @where_args);
+
+    # For the basic work query
+    push @where_query, 'work.id = s.work';
+
+    # Select works that are related to recordings for this artist
+    my $performers_query = '
+        SELECT entity1 AS work
+            FROM l_recording_work
+            JOIN recording ON recording.id = entity0
+            JOIN artist_credit_name acn
+                    ON acn.artist_credit = recording.artist_credit
+        WHERE acn.artist = ?
+    ';
+
+    my $writers_query = '
+        -- Select works that this artist is related to
+        SELECT entity1 AS work
+            FROM l_artist_work ar
+            JOIN link ON ar.link = link.id
+            JOIN link_type lt ON lt.id = link.link_type
```

Comment:
Maybe it'd be good to base this on https://github.com/metabrainz/musicbrainz-server/pull/1171 and use `@WRITER_RELATIONSHIP_GIDS` here?

Reply:
Maybe, but that seems hopelessly blocked. So I'd rather merge as is for now, and then adapt that PR to improve this.